alien4cloud-common/pom.xml

```diff
@@ -3,7 +3,7 @@
   <parent>
     <groupId>alien4cloud</groupId>
     <artifactId>alien4cloud-parent</artifactId>
-    <version>2.1.0-DEEP-1.0.0-SNAPSHOT</version>
+    <version>2.1.0-DEEP-1.2.1</version>
```

Comment:
I think we should use SNAPSHOT versions in the branches, and keep the "Release" versions only for the tags

Reply:
Is there a real world reason why we should do this?

Reply:
The master branch should contain stable but evolving code, which should be continuously built and uploaded to maven (so they kinda act like preview of the library, which dependent libraries could in turn import during their development phase to get ready for the next release).

Maven releases versions MUST not be updated in the artifact repository (as a matter of fact repositories themselves forbid by default that). Snapshots can, and that's also the reason why such convention exists.

Release artifacts (and with that I mean Final releases, Beta, RC) should be only build from tagged code, since they should be immutable as the code from where they got built. The immutability of the artifacts guarantees the stability of the code dependent to them throughout time.

Reply:
### I know what you're saying,  but the whole idea of this merge is to tag the final version in the master branch (in our case deep). I admit I was quite eager in removing the SNAPSHOT (I should have done it when merging from deep-dev to deep). DO you want me to create a separate commit in the deep-dev that changes from SNAPSHOT to RELEASE, merge it with deep, tag the version in deep, and then make a new commit to deep-dev reversing to SNAPSHOT? I mean the whole idea of the master branch is to hold releases, And when a merge is performed a tag is added (manually or automatically). the deep branch  does not accept direct commits, only merges from other branches. Just let me know how you would like to proceed.

Reply:
Hi @asalic,
I don't want to burden you too much with too many commits. Having the snapshot versions in `deep-dev` and releases in the `deep` branch will be enough for me.

Its not clear to me how you could handle eventual Betas or RCs in that way, but maybe it won't be needed anyway

Reply:
Hi @alberto-brigandi 

Okay. I will use the SNAPSHOT in the `deep-dev-UPV` and make the merges into `deep-dev` using the SNAPSHOT. Then, I will change the merged version from `deep-dev` to RELEASE (no extension) and I will merge it into the `deep` branch. Next, I will create a tag for the `deep`branch with the newly merged code. Finally, I will re-commit the SNAPSHOT tag into the  `deep-dev` branch. Let me know if this approach works for you.

Regarding the BETA, RC or other tags, I see no reason why they shouldn't be present in the `deep-dev-{insert_partner_name_here}`or `deep-dev`. The versions found in the aforementioned branches can be tagged and used If necessary. These branches are not temporary, like for a certain feature that gets merged and the branch gets deleted. They stay with the project until the termination of the repo. As a results tags can be created for them, and used when deemed necessary, especially for the `deep-dev` which represents the dev efforts of all partners.

Cheers,
A